src/ui/controls/color-picker/hsb-picker.tsx

```diff
@@ -118,15 +130,22 @@ export default function HSBPicker(props: HSBPickerProps) {
     }
 
     function onSBCanvasRender(canvas: HTMLCanvasElement) {
+        if (isElementHidden(canvas)) {
+            return;
+        }
         const hue = activeHSB.h;
         const prevHue = prevColor && rgbToHSB(parse(prevColor)).h;
-        if (hue === prevHue) {
-            return;
+        if (store.wasPrevHidden || hue !== prevHue) {
+            store.wasPrevHidden = false;
```

Comment:
Not sure if this assignment should be moved out of if...else. Just check that it works.